tests/conftest.py

```diff
@@ -274,6 +274,13 @@ def _job_supervisor_setup(request):
         PYKERN_PKDEBUG_WANT_PID_TIME='1',
         SIREPO_FEATURE_CONFIG_JOB='1',
     )
+    if 'status_test' in request.module.__name__:
```

Comment:
@robnagler env must've not been cascaded properly in a previous iteration of my work because the status_test was passing until recently. The supervisor needs to be brought up even though it wasn't until this commit. I think adding the module.__name__ test is poor abstraction (same as sbatch) but I'm not sure of a better way to do it. Thoughts?

Reply:
Odd, it seems to have not made it into master. This turns up nothing:

```sh
git log -S 'status' -- tests/conftest.py
```

The module.name approach is similar to [Aspect Oriented Programming](https://en.wikipedia.org/wiki/Aspect-oriented_programming) which I don't think should be used liberally, but at the same time, it's useful. We could add a decorator to the test case (function), which might make it more readable.

One of the bugs in pytest's fixtures is that a test can't configure them. You have to do tricks to make them configurable: have a fixture that is a wrapper that must be called to configure. I don't like any of those, which is why went with name-based configuration. 

I don't particularly like the magic trick of the fixtures being the name of the arg. I think it would simply be clearer to introduce global names from conftest.py that are prefixed with conftest or whatever so you know where they are defined. 

I think we should push ahead with what we have, and replace pytest with something less magical (more explicitly coupled).

Reply:
Thanks for the hint on AOP. 

I'll continue on as is for now. Let me know when you look into a pytest replacement. I'm curious how you evaluate the options.

Reply:
We are running into the makemake problem with pytest. About half of conftest.py is about signaling pytest to do something. conftest.py itself is not a module so at can't be tested, which was a problem. We should just expand pkunit and pykern.pkcli.test to support our testing policies. It would be work to replace the fixtures, but it would allow us to clean things up quite a bit. For example, we could import pkunit's methods (pkok, pkeq, etc.) into the test module. We wouldn't have to battle pytest's capturing of stdout and stderr. Things like work_dir would be initialized appropriately. We would enforce no imports at the global level so that we could insure appropriate initialization of flask client, pkconfig, etc.

My guess it would be a day or two of work to rewrite the parts of pytest we need, and we'd be able to remove a lot of code in pkunit, conftest, and pkcli.test.

Reply:
That makes sense. Those would be big improvements (especially the output capturing, I still run up against it sometimes).